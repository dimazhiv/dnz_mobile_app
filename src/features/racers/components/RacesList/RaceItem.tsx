import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { useOpenLink } from '../../../../hooks/hooks';
import { RaceData } from '../../../../types/RaceData';

export type RaceItemProps = {
  race: RaceData;
};

function RaceItem({ race }: RaceItemProps) {
  const { raceName, date, finalPosition, url } = race;
  const handleLinkPress = useOpenLink(url);

  return (
    <View style={styles.container}>
      <View style={styles.nameCell}>
        <Text style={styles.name}>{raceName}</Text>
      </View>
      <View style={styles.dateCell}>
        <Text>{date}</Text>
      </View>
      <View style={styles.countryCell}>
        <Text numberOfLines={1} ellipsizeMode={'tail'}>
          {finalPosition}
        </Text>
      </View>
      <View style={styles.racesCell}>
        <TouchableOpacity onPress={handleLinkPress}>
          <Text style={styles.linkStyle} numberOfLines={2}>
            info
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    borderColor: 'lightgrey',
    alignItems: 'center',
    flexDirection: 'row'
  },
  nameCell: {
    padding: 10,
    width: 100
  },
  dateCell: {
    padding: 10,
    width: 110
  },
  countryCell: {
    padding: 10,
    width: 100
  },
  racesCell: {
    padding: 10,
    minWidth: 60
  },
  name: {
    color: 'blue',
    fontWeight: 'bold'
  },
  linkStyle: {
    color: 'blue',
    textDecorationLine: 'underline'
  }
});

export default RaceItem;
