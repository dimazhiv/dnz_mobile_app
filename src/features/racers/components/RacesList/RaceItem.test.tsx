import React from 'react';
import { shallow } from 'enzyme';
import RaceItem from './RaceItem';

describe('Snapshot testing RaceItem component', () => {
  test('correct component view', () => {
    const tree = shallow(
      <RaceItem
        race={{
          date: '01.09.1994',
          raceName: 'Monza',
          finalPosition: 34,
          url: 'https://wikipedia.com'
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
