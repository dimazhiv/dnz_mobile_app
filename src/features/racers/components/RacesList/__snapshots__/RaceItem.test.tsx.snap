// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing RaceItem component correct component view 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "lightgrey",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "padding": 10,
        "width": 100,
      }
    }
  >
    <Text
      style={
        {
          "color": "blue",
          "fontWeight": "bold",
        }
      }
    >
      Monza
    </Text>
  </View>
  <View
    style={
      {
        "padding": 10,
        "width": 110,
      }
    }
  >
    <Text>
      01.09.1994
    </Text>
  </View>
  <View
    style={
      {
        "padding": 10,
        "width": 100,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
    >
      34
    </Text>
  </View>
  <View
    style={
      {
        "minWidth": 60,
        "padding": 10,
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
    >
      <Text
        numberOfLines={2}
        style={
          {
            "color": "blue",
            "textDecorationLine": "underline",
          }
        }
      >
        info
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
